test/queries/ledger_test.go

```diff
@@ -186,3 +192,187 @@ func TestLedgerBlockQueries(t *testing.T) {
 	})
 
 }
+
+// Scenario:
+// HTTP GET "/ledger/path?start={startId}&end={endId}" gets the shortest path between blocks
+// HTTP GET "/ledger/path?start={startId}&end={endId}" with invalid query params
+// where {startId} and {endId} are: out of range, reverse order
+func TestLedgerPathQueries(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     1,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	s := c.Servers[leaderIndex]
+
+	// add a few data blocks
+	var txIDs []string
+	for i := 1; i <= 41; i++ {
+		txID, rcpt, err := s.WriteDataTx(t, worldstate.DefaultDBName, fmt.Sprintf("key-%d", i), []byte{uint8(i), uint8(i)})
+		require.NoError(t, err)
+		require.Equal(t, uint64(0), rcpt.GetTxIndex())
+		txIDs = append(txIDs, txID)
+	}
+	blockRespEnv, err := s.QueryLastBlockStatus(t)
+	require.NoError(t, err)
+	require.NotNil(t, blockRespEnv)
+	blockResp := blockRespEnv.GetResponse()
+	require.NotNil(t, blockResp)
+	require.Equal(t, uint64(42), blockResp.GetBlockHeader().GetBaseHeader().GetNumber())
+
+	type testCase struct {
+		name            string
+		start           uint64
+		end             uint64
+		expectZeroBased []uint64
+		expectedErr     error
+	}
+
+	testCases := []testCase{
+		{
+			name:            "1 to 10",
+			start:           1,
+			end:             10,
+			expectZeroBased: []uint64{9, 8, 0},
+		},
+		{
+			name:            "5 to 15",
+			start:           5,
+			end:             15,
+			expectZeroBased: []uint64{14, 12, 8, 4},
+		},
+		{
+			name:            "2 to 40",
+			start:           2,
+			end:             40,
+			expectZeroBased: []uint64{39, 38, 36, 32, 16, 8, 4, 2, 1},
+		},
+		{
+			name:            "1 to 1",
+			start:           1,
+			end:             1,
+			expectZeroBased: []uint64{0},
+		},
+		{
+			name:            "6 to 6",
+			start:           6,
+			end:             6,
+			expectZeroBased: []uint64{5},
```

Comment:
having 6 instead of 5 would make readability easier

Reply:
done